### PR TITLE
Perform minor cleanup

### PR DESCRIPTION
**Pull request by sandersky**
**This project uses [semver](http://semver.org), please check the scope of this pr:**

- [x] #patch# - backwards-compatible bug fix
- [ ] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG

* Performed minor cleanup with `npm run eslint --- --fix` while running latest [eslint-config-frost-standard](https://github.com/ciena-frost/eslint-config-frost-standard) which includes new rules from [eslint-plugin-ember-standard](https://github.com/ciena-blueplanet/eslint-plugin-ember-standard) and [eslint-plugin-ocd](https://github.com/ciena-blueplanet/eslint-plugin-ocd).

### CODE REVIEW

**Review by sglanzer-fire — APPROVED** [review]
Love the fact that this was done via the fix command - stellar work!

**Comment by sglanzer-fire** [comment]
Approved 

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-core/pull-request/358/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-core)

File: addon/components/frost-select-dropdown.js [hunk 1]
```diff
@@ -91,7 +91,7 @@ export default Component.extend({
     ]
       .join(';')
 
-    return Ember.String.htmlSafe(style)
+    return EmberString.htmlSafe(style)
```
Comment (juwara0) [inline addon/components/frost-select-dropdown.js]: Since we do not have more than one String library in use right now, I would rather see us keep this as `String.htmlSafe()` and destructuring as `const {String} = Ember`.

File: addon/components/frost-select-dropdown.js [hunk 2]
```diff
@@ -108,7 +108,7 @@ export default Component.extend({
       style.push(`bottom:${bottom - ARROW_HEIGHT + BORDER_HEIGHT}px`)
     }
 
-    return Ember.String.htmlSafe(style.join(';'))
+    return EmberString.htmlSafe(style.join(';'))
```
Comment (juwara0) [inline addon/components/frost-select-dropdown.js]: Same comment as above.

File: addon/components/frost-select-li.js [hunk 3]
```diff
@@ -48,7 +48,7 @@ export default Component.extend({
       const pattern = new RegExp(filter, 'gi')
       const label = data.label.replace(pattern, '<u>$&</u>')
 
-      return Ember.String.htmlSafe(label)
+      return EmberString.htmlSafe(label)
```
Comment (juwara0) [inline addon/components/frost-select-li.js]: Same as previous comment.

File: tests/dummy/app/pods/field/controller.js [hunk 4]
```diff
@@ -1,10 +1,10 @@
 import Ember from 'ember'
 const {
-  Controller
+  Controller, inject
```
Comment (juwara0) [inline tests/dummy/app/pods/field/controller.js]: We should update this to be on one line to follow our new conventions.

File: tests/dummy/app/pods/icons/controller.js [hunk 5]
```diff
@@ -16,7 +16,7 @@ export default Controller.extend({
   iconPacks: computed('iconPacks', () => {
     return Object.keys(iconPacks).map((name) => {
       return {
-        name: Ember.String.capitalize(name),
+        name: EmberString.capitalize(name),
```
Comment (juwara0) [inline tests/dummy/app/pods/icons/controller.js]: Same comment as above about using String.

File: tests/dummy/app/pods/radio/controller.js [hunk 6]
```diff
@@ -15,7 +15,7 @@ export default Controller.extend({
     {label: 'e', value: 'e'}
   ],
 
-  model: Ember.Object.create({
+  model: EmberObject.create({
```
Comment (juwara0) [inline tests/dummy/app/pods/radio/controller.js]: I would prefer that we not do this since we only have one Object library right now.

File: tests/dummy/app/router.js [hunk 7]
```diff
@@ -1,7 +1,8 @@
-import Ember from 'ember'
 import config from './config/environment'
+import Ember from 'ember'
+const {Router: EmberRouter} = Ember
 
-var Router = Ember.Router.extend({
+var Router = EmberRouter.extend({
```
Comment (juwara0) [inline tests/dummy/app/router.js:5]: I would prefer that we not do this since we only have one Router library right now.

File: tests/integration/components/frost-password-test.js [hunk 8]
```diff
@@ -1,15 +1,15 @@
 import {expect} from 'chai'
 import {
+  $hook
```
Comment (juwara0) [inline tests/integration/components/frost-password-test.js]: We can collapse this to one line to follow the new convention.

File: tests/integration/components/frost-toggle-test.js [hunk 9]
```diff
@@ -1,15 +1,15 @@
 import {expect} from 'chai'
 import {
+  $hook
+} from 'ember-hook'
+import {
```
Comment (juwara0) [inline tests/integration/components/frost-toggle-test.js]: We can collapse these to one line to follow the new convention.

File: tests/unit/components/frost-radio-button-test.js [hunk 10]
```diff
@@ -1,17 +1,17 @@
...
+import Component from 'ember-frost-core/components/frost-component'
 import { describeComponent } from 'ember-mocha'
 import {
   beforeEach,
   describe,
   it
 } from 'mocha'
 
-import Component from 'ember-frost-core/components/frost-component'
+const {
```
Comment (juwara0) [inline tests/unit/components/frost-radio-button-test.js]: We can collapse this to one line to follow the new convention.

File: tests/unit/components/frost-radio-group-test.js [hunk 11]
```diff
@@ -1,17 +1,17 @@
...
 import {
   beforeEach,
   describe,
   it
 } from 'mocha'
 
-import Component from 'ember-frost-core/components/frost-component'
+const {
+  run
+} = Ember
```
Comment (juwara0) [inline tests/unit/components/frost-radio-group-test.js]: We can collapse this to one line.

File: tests/unit/helpers/array-test.js [hunk 12]
```diff
@@ -1,11 +1,11 @@
 import { expect } from 'chai'
 import {
+  array
+} from 'ember-frost-core/helpers/array'
+import {
```
Comment (juwara0) [inline tests/unit/helpers/array-test.js]: We can collapse these to one line.

File: tests/unit/utils/utils-test.js [hunk 13]
```diff
@@ -1,13 +1,14 @@
...
+import {cloneEvent} from 'ember-frost-core/utils'
 import {
   describe,
   it
 } from 'mocha'
-import {cloneEvent} from 'ember-frost-core/utils'
+
+const {
+  $
+} = Ember
```
Comment (juwara0) [inline tests/unit/utils/utils-test.js]: We can collapse the import and const to one line.

File: addon/helpers/ehook.js [hunk 14]
```diff
@@ -4,7 +4,7 @@
  * @see {@link https://github.com/Ticketfly/ember-hook/pull/35}
  */
 import Ember from 'ember'
-const {Helper, deprecate} = Ember
+const {Helper, deprecate, keys} = Ember
```
Comment (notmessenger) [inline addon/helpers/ehook.js:7]: Do we have guidance about whether these should be alphabetized?

File: tests/acceptance/link-test.js [hunk 15]
```diff
@@ -1,7 +1,7 @@
-import { describe, it, beforeEach, afterEach } from 'mocha'
-import { expect } from 'chai'
-import startApp from '../helpers/start-app'
 import destroyApp from '../helpers/destroy-app'
+import startApp from '../helpers/start-app'
+import { expect } from 'chai'
```
Comment (notmessenger) [inline tests/acceptance/link-test.js]: Are there supposed to be spaces before/after the braces?

File: tests/acceptance/link-test.js [hunk 16]
```diff
@@ -1,7 +1,7 @@
...
+import { afterEach, beforeEach, describe, it } from 'mocha'
```
Comment (notmessenger) [inline tests/acceptance/link-test.js]: Are there supposed to be spaces before/after the braces?

File: addon/helpers/ehook.js (hunk 14 above)
Comment (notmessenger) [reply addon/helpers/ehook.js:7]: The result of this inquiry applies equally to many other changes in this PR in other files.

File: addon/components/frost-select-dropdown.js (hunk 1 above)
Comment (sandersky) [reply addon/components/frost-select-dropdown.js]: String is a Javascript class that is globally available. If we import as String we override that which could have unforeseen side effects.

File: tests/dummy/app/pods/field/controller.js (hunk 4 above)
Comment (sandersky) [reply tests/dummy/app/pods/field/controller.js]: I agree, but that is another ESLint rule we should create (this stuff was automatically done by running `npm run eslint --- --fix`). Anything else we want done I want to create more rules for so we implement the fix once and get it everywhere for free.

File: tests/dummy/app/pods/radio/controller.js (hunk 6 above)
Comment (sandersky) [reply tests/dummy/app/pods/radio/controller.js]: Same as with `String`, `Object` is a Javascript global.

File: tests/dummy/app/router.js (hunk 7 above)
Comment (sandersky) [reply tests/dummy/app/router.js:5]: Notice the variable here is named `Router`. Either the import or this variable which is exported needs renamed.

File: addon/helpers/ehook.js (hunk 14 above)
Comment (sandersky) [reply addon/helpers/ehook.js:7]: They actually are alphabetized. When you sort an array of strings uppercase letters all come before lowercase. If we disagree with that then we need to update the sorting rules in `eslint-plugin-ocd`.

File: tests/integration/components/frost-password-test.js (hunk 8 above)
Comment (notmessenger) [reply tests/integration/components/frost-password-test.js]: Same with the `ember-mocha` line below as well?

File: tests/integration/components/frost-toggle-test.js (hunk 9 above)
Comment (notmessenger) [reply tests/integration/components/frost-toggle-test.js]: Same with the `mocha` line below?

File: tests/unit/components/frost-button-test.js [hunk 17]
```diff
@@ -1,6 +1,6 @@
 import {expect} from 'chai'
 import Ember from 'ember'
-const {run} = Ember
+import Component from 'ember-frost-core/components/frost-component'
 import {describeComponent} from 'ember-mocha'
 import {
```
Comment (notmessenger) [inline tests/unit/components/frost-button-test.js]: Should be on one line?

File: tests/unit/components/frost-checkbox-test.js [hunk 18]
```diff
@@ -1,6 +1,6 @@
 import {expect} from 'chai'
 import Ember from 'ember'
-const {$, run} = Ember
+import Component from 'ember-frost-core/components/frost-component'
 import {describeComponent} from 'ember-mocha'
 import {
```
Comment (notmessenger) [inline tests/unit/components/frost-checkbox-test.js]: Should these all be one line?

File: tests/unit/components/frost-icon-test.js [hunk 19]
```diff
@@ -1,16 +1,16 @@
 import {expect} from 'chai'
 import Ember from 'ember'
-const {run} = Ember
+import Component from 'ember-frost-core/components/frost-component'
 import {describeComponent} from 'ember-mocha'
+import PropTypeMixin from 'ember-prop-types'
+import SpreadMixin from 'ember-spread'
 import {
```
Comment (notmessenger) [inline tests/unit/components/frost-icon-test.js]: Should these all be one line?

File: tests/unit/components/frost-link-test.js [hunk 20]
```diff
@@ -1,16 +1,17 @@
 import {expect} from 'chai'
 import Ember from 'ember'
-const {run} = Ember
 import {describeComponent} from 'ember-mocha'
+import PropTypeMixin from 'ember-prop-types'
+import SpreadMixin from 'ember-spread'
 import {
```
Comment (notmessenger) [inline tests/unit/components/frost-link-test.js]: Should these all be one line?

File: tests/unit/components/frost-radio-group-test.js [hunk 21]
```diff
@@ -1,17 +1,17 @@
 /* jshint expr:true */
 import { expect } from 'chai'
 import Ember from 'ember'
-const {
-  run
-} = Ember
+import Component from 'ember-frost-core/components/frost-component'
 import { describeComponent } from 'ember-mocha'
 import {
```
Comment (notmessenger) [inline tests/unit/components/frost-radio-group-test.js]: Should these all be one line?

File: addon/helpers/ehook.js (hunk 14 above)
Comment (notmessenger) [reply addon/helpers/ehook.js:7]: yeah, didn't think about that occurrence

File: test-support/helpers/ember-frost-core.js [hunk 22]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable ember-standard/destructure */
```
Comment (notmessenger) [inline test-support/helpers/ember-frost-core.js:1]: Why is this added to some file but not others?  I have not yet discovered the pattern requiring it.

File: addon/components/frost-select-dropdown.js (hunk 1 above)
Comment (job13er) [reply addon/components/frost-select-dropdown.js]: That's the same reason we don't do
```js
const {Object} = Ember
```
Even if there are no unforseen side-effects, I think it's misleading for things like `Object`, `String`, `Number` to be overwritten within a module. You probably wouldn't want a variable called `window` for a similar reason.

File: addon/components/frost-select-dropdown.js (hunk 2 above)
Comment (job13er) [reply addon/components/frost-select-dropdown.js]: Same justification as above.

File: addon/components/frost-select-li.js (hunk 3 above)
Comment (job13er) [reply addon/components/frost-select-li.js]: Ditto.

File: tests/dummy/app/router.js (hunk 7 above)
Comment (job13er) [reply tests/dummy/app/router.js:5]: I would rename the instance personally.

File: tests/acceptance/link-test.js (hunk 15 above)
Comment (job13er) [reply tests/acceptance/link-test.js]: I would vote no. Our current eslint rule only enforces consistency, i.e. `import {expect } from 'chai'` will fail, but I've been removing the extra padding spaces when I see them (they were generated that way but in our new generators I'm not adding the extra spaces).

File: addon/components/frost-select-dropdown.js (hunk 1 above)
Comment (juwara0) [reply addon/components/frost-select-dropdown.js]: Alright, then why not just leave the invocation as `Ember.String.htmlSafe()` instead of trying to destructure and having to assign it to an alias.

Comment (sandersky) [reply addon/components/frost-select-dropdown.js]: @juwara0 one advantage is if you make a typo here:
```js
import Ember from 'ember'
const {Component, Strung: EmberString} = Ember
import computed, {readOnly} from 'ember-computed-decorators'

export default Component.extend({
  @readOnly
  @computed('title')
  header (title) {
    return EmberString.htmlSafe(`<h1>${title}</h1>`)
  }
})
```
it will be caught at compile time whereas the following typo won't:
```js
import Ember from 'ember'
const {Component} = Ember
import computed, {readOnly} from 'ember-computed-decorators'

export default Component.extend({
  @readOnly
  @computed('title')
  header (title) {
    return Ember.Strung.htmlSafe(`<h1>${title}</h1>`)
  }
})
```

File: tests/acceptance/link-test.js (hunk 15 above)
Comment (sandersky) [reply tests/acceptance/link-test.js]: I would vote no as well but this is yet to be an ESLint in our stack so until it is shouldn't block PR's.

File: tests/dummy/app/router.js (hunk 7 above)
Comment (juwara0) [inline tests/dummy/app/router.js]: Or we just leave this as `Ember.Router`.

File: tests/acceptance/link-test.js (hunk 15 above)
Comment (sandersky) [reply tests/acceptance/link-test.js]: If we all agree then we can simply add http://eslint.org/docs/rules/object-curly-spacing to eslint-config-frost-standard.

File: addon/components/frost-select-dropdown.js (hunk 1 above)
Comment (juwara0) [reply addon/components/frost-select-dropdown.js]: I am not crazy about the `EmberString`, `EmberObject` and `EmberRouter` approach. It feels forced just so that we can destructure but then the need to use an alias makes it less clean. I think that in these cases it is more clear to just use `Ember.String`, `Ember.Object` and `Ember.Router`.

Comment (job13er) [reply addon/components/frost-select-dropdown.js]: I agree with the first two, the second should never be needed, there shouldn't be another `Router` symbol in that file, it should be renamed. 

I believe the agreed upon approach here is to begin a discussion in `#frost-desgin` if we want to move away from the previous agreed-upon convention of descructuring everything *except* what clashes with a built-in and leaving those on the `Ember` object `Ember.String`, `Ember.Object` but *not* `Ember.Router`, in that case, the `var Router` should be changed (in case I didn't say that enough times yet ;)

File: test-support/helpers/ember-frost-core.js (hunk 22 above)
Comment (sandersky) [reply test-support/helpers/ember-frost-core.js:1]: The new destructure rule wants all Ember classes destructured except String and Object. In the case of this file it does:
```js
const assign = Object.assign || Ember.assign || Ember.merge
```
in order to support older versions of Ember. Without this rule disabled it would complain that `Ember.assign` and `Ember.merge` are not destructured.

File: tests/integration/components/frost-checkbox-test.js [hunk 23]
```diff
@@ -248,7 +245,7 @@ describeComponent(
     })
 
     // https://github.com/juwara0/ember-frost-core/issues/1
-    it.skip('sets focus on render when autofocus is true', function () {
+    it('sets focus on render when autofocus is true', function () {
```
Comment (juwara0) [inline tests/integration/components/frost-checkbox-test.js]: The issue that is causing this test to fail is not fixed yet: https://github.com/ciena-frost/ember-frost-core/issues/257 

This test was skipped until the issue was fixed.

**Comment by sglanzer-fire** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-core/pull-request/358/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-core)
